src/Chainweb/CutDB.hs

```diff
@@ -481,110 +480,64 @@ startCutDb config logger headerStore providers cutHashesStore = mask_ $ do
                 blockHeaders <- iforM rankedBlockHashes (lookupRankedWebBlockHeaderDb wbhdb)
                 return $ unsafeMkCut blockHeaders
 
+-- | Sync all configured payload providers with consensus.
+--
 synchronizeProviders
     :: (Logger logger, HasVersion)
-    => logger -> WebBlockHeaderDb -> ChainMap ConfiguredPayloadProvider -> Cut -> IO Cut
+    => logger
+    -> WebBlockHeaderDb
+    -> ChainMap ConfiguredPayloadProvider
+    -> Cut
+    -> IO Cut
 synchronizeProviders logger wbh providers c = do
     let startHeaders = HM.unionWith
-            (\startHeader _genesisHeader -> startHeader)
```

Comment:
this should be equal to `HM.union (_cutHeaders c) $ (imap (\cid () -> genesisBlockHeader cid) (HS.toMap chainIds)`